visionSamples/barcode-reader/app/src/main/java/com/google/android/gms/samples/vision/barcodereader/BarcodeTrackerFactory.java

```diff
@@ -24,17 +26,22 @@
  * Factory for creating a tracker and associated graphic to be associated with a new barcode.  The
  * multi-processor uses this factory to create barcode trackers as needed -- one for each barcode.
  */
-class BarcodeTrackerFactory implements MultiProcessor.Factory<Barcode> {
-    private GraphicOverlay<BarcodeGraphic> mGraphicOverlay;
+public class BarcodeTrackerFactory implements MultiProcessor.Factory<Barcode> {
```

Comment:
Please keep it package-private if possible.

Reply:
In order to keep a project's classes organized the BarcodeTrackerFactory is public so it can live in a separate package for the Camera utilities and then the Activities / Fragments can access these methods from a separate package. 

In the sample below, the _BarcodeActivity_ implements _BarcodeGraphicTracker_ and _BarcodeTrackerFactory_ from a parent package which is why they are public.

![screen shot 2017-09-22 at 9 53 22 am](https://user-images.githubusercontent.com/3938076/30755535-f3aec2f0-9f7b-11e7-8d11-abb4a352f9f9.png)